server/src/main/kotlin/com/kamelia/hedera/rest/user/DiskQuotaService.kt

```diff
@@ -0,0 +1,75 @@
+package com.kamelia.hedera.rest.user
+
+import com.kamelia.hedera.database.Connection
+import com.kamelia.hedera.rest.auth.SessionManager
+import com.kamelia.hedera.util.uuid
+import com.kamelia.hedera.util.withReentrantLock
+import java.sql.Connection.TRANSACTION_REPEATABLE_READ
+import java.util.*
+import kotlinx.coroutines.sync.Mutex
+
+object DiskQuotaService {
+
+    private val quotas = mutableMapOf<UUID, Pair<Long, Long>>()
+    private val mutex = Mutex()
+
+    private suspend fun User.getDiskQuotaOrInsert(): Pair<Long, Long> {
+        if (!quotas.containsKey(id.value)) {
+            quotas[id.value] = Connection.transaction(transactionIsolation = TRANSACTION_REPEATABLE_READ) {
+                repetitionAttempts = 10
+                minRepetitionDelay = 500
+
+                currentDiskQuota to maximumDiskQuota
+            }
+        }
+        return quotas[id.value]!!
+    }
+
+    private suspend fun User.saveQuota() {
```

Comment:
make saveQuota receive the quota as a parameter. This will avoid regetting from the map everytime, even though you're calling saveQuota just after setting. Dont forget to update the call sites, and avoid setting and getting (only set)